sdk/go/src/sawtooth_sdk/client/signer.go

```diff
@@ -0,0 +1,106 @@
+package client
+
+import (
+	"crypto/sha256"
+	"crypto/sha512"
+	ellcurv "github.com/btcsuite/btcd/btcec"
+	"math/big"
+)
+
+// -- Key Gen --
+
+// GenPrivKey generates a new private key and returns it as bytes
+func GenPrivKey() []byte {
+	priv, err := ellcurv.NewPrivateKey(ellcurv.S256())
+	if err != nil {
+		panic("Failed to generate private key")
+	}
+	return priv.Serialize()
+}
+
+// GenPubKey generates a new public key from a given private key and returns it
+// using the 33 byte compressed format
+func GenPubKey(privkey []byte) []byte {
+	_, pub := ellcurv.PrivKeyFromBytes(ellcurv.S256(), privkey)
+	return pub.SerializeCompressed()
```

Comment:
Is there some utility in giving users the raw binary instead of a hex string? I see you are encoding it for them during transaction creation, but is a binary going to be the easiest format to store or transmit? Will people be confused by having a public key that is a binary when the documentation shows hex strings?